### PR TITLE
HPACK dynamic table tests in http/2 parser

### PR DESCRIPTION
**Pull request by alegz-filomaphitskay**
Contributes to #1297.

What was done here:
- added tests for hpack dynamic table;
- in do_split_and_parse() HPACK context is saved before running request's parsing and it is restored before using next chunk size.

### CODE REVIEW

File: fw/t/unit/test_http_parser_common.h [hunk 1]
```diff
@@ -461,12 +463,19 @@ do_split_and_parse(int type, int chunk_mode)
...
 	}
 	else if (type == FUZZ_REQ_H2) {
-		if (req)
-			test_req_free(req);
+		static TfwH2Ctx	h2_origin;
+
+		if (chunk_size_index == 0)
+			h2_origin = conn.h2;
+		else
+			conn.h2 = h2_origin;
```
Comment (krizhanovsky) [inline fw/t/unit/test_http_parser_common.h:471]: Please write a comment why do we need to save and restore HTTP/2 context for each request processing

**Review by krizhanovsky — APPROVED** [review]
Quite good test suite and I'm very happy that it was possible to make the tests for the dynamic table! Just please adjust the coding style before merging

File: fw/t/unit/test_http_parser_common.h (hunk 1 above)
Comment (alegz-filomaphitskay) [reply fw/t/unit/test_http_parser_common.h:471]: Done.

File: fw/t/unit/test_http2_parser_hpack.c [hunk 2]
```diff
@@ -21,10 +21,349 @@
...
+	(strlen(data) + HPACK_ENTRY_OVERHEAD)
+
+
+TEST(http2_parser_hpack, literal_header_field_with_incremental_indexing)
+{
+	const char *s_dummy = "dummy" "super123";
+
+	EXPECT_BLOCK_REQ_H2_HPACK(
+	    HEADERS_FRAME_BEGIN();
+		// :method = "GET"
```
[the remark this answers is not in the document]
Comment (alegz-filomaphitskay) [reply fw/t/unit/test_http2_parser_hpack.c:60]: Done.

**Pull request merged**